packages/powersync_attachments_stream/lib/src/local_storage.dart

```diff
@@ -0,0 +1,84 @@
+import 'dart:convert';
+import 'dart:io';
+import 'package:path/path.dart' as p;
+
+abstract class LocalStorage {
+  /// Saves binary data to storage with an identifier
+  /// 
+  /// [id] - Unique identifier for the stored data
+  /// [bytes] - Binary data to store
+  /// [mediaType] - Optional MIME type of the data (e.g., 'image/jpeg')
+  /// [metadata] - Optional key-value pairs for additional data information
+  Future<void> save(String id, List<int> bytes, {String? mediaType, Map<String, dynamic>? metadata});
+  
+  /// Retrieves binary data by identifier
+  /// 
+  /// [id] - Unique identifier of the data to retrieve
+  /// Returns the binary data if found, null otherwise
+  Future<List<int>?> read(String id);
+  
+  /// Removes data and its metadata from storage
+  /// 
+  /// [id] - Unique identifier of the data to delete
+  Future<void> delete(String id);
+  
+  /// Retrieves metadata associated with stored data
+  /// 
+  /// [id] - Unique identifier of the data
+  /// Returns metadata map if found, null otherwise
+  Future<Map<String, dynamic>?> getMetadata(String id);
+}
+
+class IOLocalStorage implements LocalStorage {
```

Comment:
I think it would be nice if we could move code that depends on `dart:io` into different files.

Then, on the public level (outside of `src/`), we could have an import for users who want to use this on the web (writing their own local storage implementations for e.g. the web filesystem APIs, but being unable to import `dart:io`).
So there could be e.g. a `package:powersync_attachments_stream/powersync_attachments_stream.dart` for the default things that depend on `dart:io`, and another `package:powersync_attachments_stream/common.dart` for the subset of files that work on all platforms.